manuscript/template.tex

```diff
@@ -0,0 +1,667 @@
+	\label{fig:uml_datastructure}
+    \end{figure}   
+\unskip
+
+\subsection{Document Generation}
+As noted multiple times throughout this article, there have been a number of figures and tables generated from the YAML files placed within the SeaLion mission architecture GitHub repository.  Many of the figures are UML diagrams that are auto-generated artifacts rendered from the M30ML modeling language and formatted using the Liquid template language.  This is the link for the docs-as-code approach.  YAML code files are used to generate documents for information sharing between group members.  This means that any changes made to the SeaLion mission architecture model can immediately be used to generate new documents.  Whether it be diagrams, tables, or text, continuous updating is ensured that any changes affecting dependencies within the mission architecture are kept in sync.  A conference proceeding manuscript presented in AIAA SciTech 2023 was created purely by a docs-as-code format \cite{scitech_proceeding}.  The team used a LaTeX template to automatically format the manuscript to the conference guidelines and subsequently inject items such as the generated diagrams, tables, and references directly into the manuscript.
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\section{Component Implementation}
+The implementation of M30ML for components is still a work in progress at the time of this article's publication.  A brief description of current component implementation into the SeaLion model is provided.  However, many of the implementation methods mentioned herein are subjected to change and are not finalized unlike the architecture described in the prior section.
+
+\subsection{Distributed OSHW Framework}
+The M30ML pillars are based on Open Source Hardware (OSHW) principals \cite{mach30_git}.  The current SeaLion mission architecture repository, at time of article publication, for components is “structured as a Distributed OSHW Framework (DOF) - component for defining the contents of the Mission concept of operations (ConOps) as a collection of nested subcomponents, component interfaces, and component functions for generating bill of materials (BOMs) and assembly instructions for the SeaLion CubeSat” \cite{sealion_mission_architecture}.  Inside the components folder of the SeaLion mission architecture repository there are two subfolders; one labeled with 'sealion-cubesat' and another labeled with 'sealion-ground-station'.  Each of those folders would contain a components folder and subsequently those individual labeled components can have their own components folder.  Thus, a chain of components and subcomponents can be created as illustrated in figure \ref{fig:components_file}.  A parts YAML file in each components folder details what the subcomponents would be.  An excerpt example for the main SeaLion cubesat is provided in figure \ref{fig:components} that is associated with the file folders shown in figure \ref{fig:components_file}.
```

Comment:
`M30ML pillars` should be `DOF pillars`

Reply:
Resolved bbd10cadce0dc6665d7e685b218b57c669172dc5